src/taipy/core/data/sql.py

```diff
@@ -165,57 +165,41 @@ def _read_as_pandas_dataframe(self, columns: Optional[List[str]] = None):
             return pd.read_sql_query(self.read_query, con=self.__engine())[columns]
         return pd.read_sql_query(self.read_query, con=self.__engine())
 
+    def _create_table(self, engine) -> Table:
+        return Table(
+            self.write_table,
+            MetaData(),
+            autoload=True,
+            autoload_with=engine,
+        )
+
     def _write(self, data) -> None:
         """Check data against a collection of types to handle insertion on the database."""
-        with self.__engine().connect() as connection:
-            write_table = table(self.write_table)
-            if isinstance(data, pd.DataFrame):
-                self.__insert_dicts(data.to_dict(orient="records"), write_table, connection)
-            elif isinstance(data, dict):
-                self.__insert_dicts([data], write_table, connection)
-            elif isinstance(data, tuple):
-                self.__insert_tuples([data], write_table, connection)
-            elif isinstance(data, list) or isinstance(data, np.ndarray):
-                if isinstance(data[0], tuple):
-                    self.__insert_tuples(data, write_table, connection)
-                elif isinstance(data[0], dict):
-                    self.__insert_dicts(data, write_table, connection)
-                else:
-                    self.__insert_list(data, write_table, connection)
-                    return
-            else:
-                # if data is a single primitive value (int, str, etc), pass it as a list of tuples
-                # with only one value
-                self.__insert_tuples([(data,)], write_table, connection)
 
-    @staticmethod
-    def __insert_list(data: Collection[List], write_table: Any, connection: Any) -> None:
-        """
-        :param data: a list of values
-        :param write_table: a SQLAlchemy object that represents a table
-        :param connection: a SQLAlchemy connection to write the data
-
-        This method will lookup the length of the first object of the list and build the insert through
-        creation of a string of '?' equivalent to the length of the element. The '?' character is used as
-        placeholder for a tuple of same size.
-        """
-        with connection.begin() as transaction:
-            try:
-                markers = ",".join(["(?)"] * len(data))
-                markers = markers
-                ins = "INSERT INTO {tablename} VALUES {markers}"
-                ins = ins.format(tablename=write_table.name, markers=markers)
-                connection.execute(ins, list(data))
-            except:
-                transaction.rollback()
-                raise
+        engine = self.__engine()
+
+        if isinstance(data, pd.DataFrame):
+            data = data.to_dict(orient="records")
+        elif isinstance(data, np.ndarray):
+            data = data.tolist()
+        if not isinstance(data, list):
+            data = [data]
+        if len(data) == 0:
```

Comment:
If data is None, this check will pass and we will write in the table. Is that the desired behavior?

Reply:
Yes, it is inserting 1 row of (null) into the table.